forum/users/views.py

```diff
@@ -1,11 +1,108 @@
+from os import environ
+
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.shortcuts import get_object_or_404
 from rest_framework_simplejwt.views import (
     TokenObtainPairView as BaseTokenObtainPairView,
-    TokenRefreshView as BaseTokenRefreshView
+    TokenRefreshView as BaseTokenRefreshView,
+)
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.exceptions import NotFound
+
+from users.models import CustomUser
+from forum.tasks import send_email_task
+from forum.utils import build_email_message
+from .models import PasswordResetModel
+from .serializers import (
+    PasswordResetRequestSerializer,
+    PasswordResetSerializer
 )
+from .throttling import PasswordResetThrottle
+
 
 class TokenObtainPairView(BaseTokenObtainPairView):
     throttle_scope = 'token_obtain'
 
 
 class TokenRefreshView(BaseTokenRefreshView):
     throttle_scope = 'token_refresh'
+
+
+class PasswordResetRequestView(GenericAPIView):
+    serializer_class = PasswordResetRequestSerializer
+    permission_classes = []
+    throttle_classes = [PasswordResetThrottle]
+
+    def post(self, request: Request):
+        serializer_data = self.serializer_class(data=request.data)
+        serializer_data.is_valid(raise_exception=True)
+
+        clear_data = serializer_data.validated_data
+
+        user: CustomUser = get_object_or_404(CustomUser, email=clear_data['email'])
+
+        token_generator = PasswordResetTokenGenerator()
+        reset_token = token_generator.make_token(user)
+
+        PasswordResetModel.objects.create(
+            email=clear_data['email'],
+            reset_token=reset_token
+        )
```

Comment:
It would be great to add a check if recovery token already exists for a user. If several requests were sent so that only the most recent link is active.

Reply:
But what if the email-sending task failed? So a user could be unnabled to reset the password.